engine/src/test/java/io/camunda/zeebe/engine/processing/resource/ResourceDeletionMultiPartitionTest.java

```diff
@@ -0,0 +1,101 @@
+import java.util.stream.Collectors;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestWatcher;
+
+public class ResourceDeletionMultiPartitionTest {
+
+  private static final int PARTITION_COUNT = 3;
+  private static final String DMN_RESOURCE = "/dmn/decision-table.dmn";
+
+  @Rule public final EngineRule engine = EngineRule.multiplePartition(PARTITION_COUNT);
+  @Rule public final TestWatcher watcher = new RecordingExporterTestWatcher();
+
+  @Test
+  public void shouldTestDmnLifecycle() {
+    // given
+    engine.deployment().withXmlClasspathResource(DMN_RESOURCE).deploy();
+
+    final var resourceKey =
+        RecordingExporter.decisionRequirementsRecords()
+            .withIntent(DecisionRequirementsIntent.CREATED)
+            .withDecisionRequirementsId("force_users")
+            .getFirst()
+            .getKey();
+
+    // when
+    engine.resourceDeletion().withResourceKey(resourceKey).delete();
+
+    // then
+    final var index = new AtomicInteger();
+    assertThat(
+            RecordingExporter.records()
+                .withPartitionId(1)
+                .limit(
+                    r ->
+                        r.getIntent().equals(ResourceDeletionIntent.DELETED)
+                            && index.incrementAndGet() == 3))
+        .extracting(
+            Record::getIntent,
+            Record::getRecordType,
+            r ->
+                // We want to verify the partition id where the deletion was distributing to and
+                // where it was completed. Since only the CommandDistribution records have a
+                // value that contains the partition id, we use the partition id the record was
+                // written on for the other records.
+                r.getValue() instanceof CommandDistributionRecordValue
+                    ? ((CommandDistributionRecordValue) r.getValue()).getPartitionId()
+                    : r.getPartitionId())
+        .containsSubsequence(
+            tuple(ResourceDeletionIntent.DELETE, RecordType.COMMAND, 1),
+            tuple(DecisionIntent.DELETED, RecordType.EVENT, 1),
+            tuple(DecisionRequirementsIntent.DELETED, RecordType.EVENT, 1),
+            tuple(ResourceDeletionIntent.DELETED, RecordType.EVENT, 1),
+            tuple(CommandDistributionIntent.STARTED, RecordType.EVENT, 1))
+        .containsSubsequence(
+            tuple(CommandDistributionIntent.DISTRIBUTING, RecordType.EVENT, 2),
+            tuple(CommandDistributionIntent.ACKNOWLEDGE, RecordType.COMMAND, 2),
+            tuple(CommandDistributionIntent.ACKNOWLEDGED, RecordType.EVENT, 2))
+        .containsSubsequence(
+            tuple(CommandDistributionIntent.DISTRIBUTING, RecordType.EVENT, 3),
+            tuple(CommandDistributionIntent.ACKNOWLEDGE, RecordType.COMMAND, 3),
+            tuple(CommandDistributionIntent.ACKNOWLEDGED, RecordType.EVENT, 3))
+        .endsWith(tuple(CommandDistributionIntent.FINISHED, RecordType.EVENT, 1));
```

Comment:
🔧 I understand that commands appear to be part of the lifecycle, but I'd prefer asserting events only. Events are what really happened, commands are just our implementation to get there. Commands can also be rejected.